src/js/components/ServiceInfo.js

```diff
@@ -7,7 +8,73 @@ import PageHeader from './PageHeader';
 import Service from '../structs/Service';
 import StringUtil from '../utils/StringUtil';
 
+const ServiceActionItem = {
+  EDIT: 'edit',
+  DESTROY: 'destroy',
+  SCALE: 'scale',
+  SUSPEND: 'suspend',
+  MORE: 'more'
+};
+
 class ServiceInfo extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      isServiceFormModalShown: false
+    };
+
+    this.onActionsItemSelection =
+      this.onActionsItemSelection.bind(this);
+  }
+
+  onActionsItemSelection(item) {
+    if (item.id === ServiceActionItem.EDIT) {
+      this.setState({
+        isServiceFormModalShown: true
+      });
+    }
+  }
+
+  getActionButtons() {
+    const dropdownItems = [{
+      className: 'hidden',
+      id: ServiceActionItem.MORE,
+      html: '',
+      selectedHtml: 'More'
+    }, {
+      id: ServiceActionItem.SCALE,
+      html: 'Scale'
+    }, {
+      id: ServiceActionItem.SUSPEND,
+      html: 'Suspend'
+    }, {
+      id: ServiceActionItem.DESTROY,
+      html: <span className="text-danger">Destroy</span>
+    }];
+
+    return [(
+      <button className="button button-stroke button-inverse flush-bottom"
+        key="action-button-edit"
+        onClick={() =>
+          this.onActionsItemSelection({id: ServiceActionItem.EDIT})}>
+        Edit
+      </button>
+    ), (
+      <Dropdown
+        key="actions-dropdown"
+        buttonClassName="button button-stroke button-inverse dropdown-toggle"
+        dropdownMenuClassName="dropdown-menu"
+        dropdownMenuListClassName="dropdown-menu-list"
+        dropdownMenuListItemClassName="clickable"
+        wrapperClassName="dropdown"
```

Comment:
You can add `anchor-right` to the `wrapperClassName` prop which will prevent it from running off the screen